x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/components/cytoscape_options.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import cytoscape from 'cytoscape';
+import theme from '@elastic/eui/dist/eui_theme_light.json';
+import {
+  ANALYSIS_CONFIG_TYPE,
+  JOB_MAP_NODE_TYPES,
+} from '../../../../../../common/constants/data_frame_analytics';
+import classificationJobIcon from './icons/ml_classification_job.svg';
+import outlierDetectionJobIcon from './icons/ml_outlier_detection_job.svg';
+import regressionJobIcon from './icons/ml_regression_job.svg';
+
+const lineColor = '#C5CCD7';
+// @ts-expect-error `documentMode` is not recognized as a valid property of `document`.
+const isIE11 = !!window.MSInputMethodContext && !!document.documentMode;
+
+enum MAP_SHAPES {
+  ELLIPSE = 'ellipse',
+  RECTANGLE = 'rectangle',
+  DIAMOND = 'diamond',
+}
+
+function shapeForNode(el: cytoscape.NodeSingular): MAP_SHAPES {
+  const type = el.data('type');
+  switch (type) {
+    case JOB_MAP_NODE_TYPES.ANALYTICS:
+      return MAP_SHAPES.ELLIPSE;
+    case JOB_MAP_NODE_TYPES.TRANSFORM:
+      return MAP_SHAPES.RECTANGLE;
+    case JOB_MAP_NODE_TYPES.INDEX_PATTERN:
+      return MAP_SHAPES.DIAMOND;
+    default:
+      return MAP_SHAPES.ELLIPSE;
+  }
+}
+
+function iconForNode(el: cytoscape.NodeSingular) {
+  const type = el.data('analysisType');
+
+  switch (type) {
+    case ANALYSIS_CONFIG_TYPE.OUTLIER_DETECTION:
+      return outlierDetectionJobIcon;
+    case ANALYSIS_CONFIG_TYPE.CLASSIFICATION:
+      return classificationJobIcon;
+    case ANALYSIS_CONFIG_TYPE.REGRESSION:
+      return regressionJobIcon;
+    default:
+      return undefined;
+  }
+}
+
+function borderColorForNode(el: cytoscape.NodeSingular) {
+  if (el.selected()) {
+    return theme.euiColorPrimary;
+  }
+
+  const type = el.data('type');
+
+  switch (type) {
+    case JOB_MAP_NODE_TYPES.ANALYTICS:
+      return theme.euiColorSecondary;
+    case JOB_MAP_NODE_TYPES.TRANSFORM:
+      return theme.euiColorVis1;
+    case JOB_MAP_NODE_TYPES.INDEX_PATTERN:
+      return theme.euiColorVis2;
+    default:
+      return theme.euiColorMediumShade;
+  }
+}
+
+export const cytoscapeOptions: cytoscape.CytoscapeOptions = {
+  autoungrabify: true,
+  boxSelectionEnabled: false,
+  maxZoom: 3,
+  minZoom: 0.2,
+  // @ts-ignore
```

Comment:
quite some `@ts-ignore` here 🤔 . is it possible to avoid it? or at least specify the reason